### PR TITLE
refactor chartoperator resource

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards Node Pools. The `chartoperator` resource will soon be dropped from `cluster-operator` anyway but for completeness in the scope of our story we need to port it as well for now.

### CODE REVIEW

File: service/controller/clusterapi/v18/cluster_resource_set.go [hunk 1]
```diff
@@ -126,6 +128,28 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 	//	}
 	//}
 
+	//var chartOperatorResource controller.Resource
```
Comment (MarcelMue) [inline service/controller/clusterapi/v18/cluster_resource_set.go]: Why is this commented out? I don't understand right now.

Comment (xh3b4sd) [reply service/controller/clusterapi/v18/cluster_resource_set.go]: There is no cool seamless migration path for moving the resources. I am just rewriting all of them and once all are in place I activate them all together and make sure this replaces the aws controller magic as a whole.

Comment (MarcelMue) [reply service/controller/clusterapi/v18/cluster_resource_set.go]: Okay, my concern here is that we don't have a gradual transition and the changes here are quite big. 

Removing / altering the deletion logic makes sense but what was the idea to have it in the first place?
There is no way to check right now if this works "in reality" which has me a bit concerned.

Comment (xh3b4sd) [reply service/controller/clusterapi/v18/cluster_resource_set.go]: I will make sure it works. No worries. Reasons for why the implementation was how it was are probably manifold and all legacy related. I do not want to get into too much detail of the why though.

**Comment by xh3b4sd** [comment]
I am going with this now and move on. Let me know when there is something to change or align.

**Pull request merged**